wordpress-mobile/WordPress-iOS
# Issues/4035 update reader topic models

Refs #4035 
Replaces the ReaderTopics model with new more focused models.  This lays the ground work for supporting different kinds of "topics" in the reader, and allows for storing more information about individual topic types. 

To Test:
- Install a copy of 5.6 first, then upgrade to this branch.  Make sure the upgrade works properly. 
- Make sure you can various kinds of topics in the reader.
- Make sure you can follow / unfollow topics
- Make sure the reader shows the Freshly Pressed and the default tag list when not logged into WordPress.com

@astralbodies Would you be up for reviewing this one? 

Needs Review: @astralbodies
